File: frontend/src/types/contentPipeline.ts
/** 多平台内容分发流水线类型定义 */

export interface PlatformAccount {
  id: string
  platform: string
  account_name: string
  account_id?: string
  follower_count: number
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface PublishSchedule {
  id: string
  title: string
  description?: string
  platforms: Array<{platform: string; account_id?: string; status: string}>
  scheduled_at: string
  is_recurring: boolean
  recurring_pattern?: string
  status: string
  published_at?: string
  error_message?: string
}

export interface SimulateResult {
  platform: string
  platform_name: string
  recommended_cover: string
  max_tags: number
  title_adapted: string
  tags_count: number
  tags_ok: boolean
}

export interface PublishStats {
  total_schedules: number
  scheduled: number
  published: number
  failed: number
  recent_7d_success: number
}

File: backend/app/schemas/content_pipeline.py
"""多平台内容分发流水线 Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PlatformAccountCreate(BaseModel):
    platform: str
    account_name: str
    account_id: Optional[str] = None
    follower_count: int = 0


class PlatformAccountResponse(BaseModel):
    id: str
    platform: str
    account_name: str
    account_id: Optional[str] = None
    follower_count: int
    is_active: bool
    created_at: str
    updated_at: str


class ScheduleCreate(BaseModel):
    title: str
    description: Optional[str] = None
    work_id: Optional[str] = None
    platforms: list[dict]
    scheduled_at: str
    is_recurring: bool = False
    recurring_pattern: Optional[str] = None


class PublishScheduleResponse(BaseModel):
    id: str
    title: str
    description: Optional[str] = None
    platforms: list[dict]
    scheduled_at: str
    is_recurring: bool
    recurring_pattern: Optional[str] = None
    status: str
    published_at: Optional[str] = None
    error_message: Optional[str] = None


class SimulateResult(BaseModel):
    platform: str
    platform_name: str
    recommended_cover: str
    max_tags: int
    title_adapted: str
    tags_count: int
    tags_ok: bool


class PublishStats(BaseModel):
    total_schedules: int
    scheduled: int
    published: int
    failed: int
    recent_7d_success: int
